operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/controller/ControllerResourceEventSource.java

```diff
@@ -103,6 +103,6 @@ private void handleKubernetesClientException(Exception e) {
 
   @Override
   public Optional<T> getSecondaryResource(T primary) {
-    return get(ResourceID.fromResource(primary));
+    throw new IllegalStateException("This method should not be called here. Primary: " + primary);
```

Comment:
Why this change?

Reply:
It does not makes sense imo to ask for the secondary resource from the `ControllerEventSource`. It basically does not have secondary, so this operation should fail.